apps/www/middleware.ts

```diff
@@ -233,10 +234,46 @@ async function middlewareFunc(req: NextRequest): Promise<NextResponse> {
     }
   }
 
+  /**
+   * Redirect based on the IP jwt-token after it was successfully validated
+   * @param token JWT found in the cookie header
+   * @returns NextResponse
+   */
+  async function rewriteBasedOnIPAllowlistToken(
+    token: string,
+  ): Promise<NextResponse> {
+    try {
+      // Parse and verify JWT to decide about redirection
+      const jwtBody = await verifyJWT(token)
+      const clientIP = jwtBody?.ip
+      const isAllowedIP =
+        clientIP &&
+        process.env.IP_ALLOWLIST &&
+        process.env.IP_ALLOWLIST.includes(clientIP)
```

Comment:
Question: would there be only one IP address allowed per customer? do we need to support wildcards and/or ranges here?

Reply:
Good question. Currently we have one request by Swissconsortium that uses one specific IP per customer. I would say we cross the ip-ranges and wildcards if there is a need for it.